lib/misc/blocks.js

```diff
@@ -102,32 +102,48 @@ export function moveNext (ed, sel, range) {
   return sel.setBufferRange([[to, Infinity], [to, Infinity]])
 }
 
-function getRanges (ed) {
-  const ranges = ed.getSelections().map((sel) => ({
-    selection: sel,
-    range: sel.isEmpty() ? getRange(ed, sel.getHeadBufferPosition().row) :
-                           getSelection(ed, sel)
-  }));
-  return ranges.filter(({range}) => (range != null) && ed.getTextInBufferRange(range).trim())
+function getRanges (ed, row) {
+  const ranges = row ?
+    [{
+      selection: undefined,
+      range: getRange(ed, row)
+    }] :
+    ed.getSelections().map(sel => {
+      return {
+        selection: sel,
+        range: sel.isEmpty() ?
+          getRange(ed, sel.getHeadBufferPosition().row) :
+          getSelection(ed, sel)
+      }
+    })
+  return ranges.filter(({ range }) => {
+    return range != null && ed.getTextInBufferRange(range).trim()
+  })
 }
 
-export function get (ed) {
-  const result = []
-  for (let {range, selection} of getRanges(ed)) {
-    result.push({
+export function get (ed, row) {
+  return getRanges(ed, row).map(({ range, selection }) => {
+    return {
       range,
       selection,
       line: range[0][0],
       text: ed.getTextInBufferRange(range)
-    })
+    }
+  })
+}
+
+export function getLocalContext (editor, row) {
+  const block = get(editor, row)[0]
```

Comment:
Seems like the `get(ed, row)`  and `getRanges(ed, row)` functionality is only used here, so maybe it'd be better to move that logic here and use `getRange` directly?

Reply:
Actually `get(ed, row)` is used in [completions.js](https://github.com/JunoLab/atom-julia-client/blob/master/lib/runtime/completions.js#L59-L62), thus I refrained from migrating `get` and `getRanges` into `getLocalContext`.

But if you don't mind the small refactor in completions.js, I will do so :)

Reply:
Ah also [evaluation.coffee](https://github.com/JunoLab/atom-julia-client/blob/master/lib/runtime/evaluation.coffee#L32) uses `get`.

So... I feel like I want to keep `get` and `getRanges` as is ?

Reply:
Mh? `completions.js` and `evaluation.coffee` only use `get(ed)`.  I don't care too much about this, so go with whatever you want ;)

Reply:
yeah, you're right. I always get confused with function calls in JS ...

Reply:
okay, maybe 8122f31 will simplify the logic